test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -705,9 +705,27 @@ private void wipeCluster() throws Exception {
             deleteAllAutoFollowPatterns();
         }
 
+        deleteAllNodeShutdownMetadata();
+
         assertThat("Found in progress snapshots [" + inProgressSnapshots.get() + "].", inProgressSnapshots.get(), anEmptyMap());
     }
 
+    /**
+     * If any nodes are registered for shutdown, removes their metadata.
+     */
+    private static void deleteAllNodeShutdownMetadata() throws IOException {
+        Request getShutdownStatus = new Request("GET", "_nodes/shutdown");
+        Map<String, Object> statusResponse = responseAsMap(client().performRequest(getShutdownStatus));
+        List<Map<String, Object>> nodesArray = (List<Map<String, Object>>) statusResponse.get("nodes");
+        List<String> nodeIds = nodesArray.stream()
+            .map(nodeShutdownMetadata -> (String) nodeShutdownMetadata.get("node_id"))
+            .collect(Collectors.toUnmodifiableList());
+        for (String nodeId : nodeIds) {
```

Comment:
This isn't part of the stream because `client().performRequest()` throws a checked exception which makes using it in a stream a pain in the butt.